src/f4enix/input/MCNPinput.py

```diff
@@ -229,6 +261,50 @@ def __init__(
         # cells.extend(surfs)
         # cells.extend(data)
 
+    @property
+    def cells(self) -> dict[str, parser.Card]:
+        """Get the cells attribute."""
+        return self._cells
+
+    @cells.setter
+    def cells(self, value: dict[str, parser.Card]) -> None:
+        """Set the cells attribute."""
+        # enforce that the keys are all corresponding to the cell number
+        # in case of difference, the dictionary key takes precedence
+        for key, card in value.items():
+            if key != str(card.name):
+                logging.warning(
+                    f"The cell number {card.name} is different from Key {key} and will be replaced"
+                )
+                card.name = int(key)
+                card._set_value_by_type("cel", int(key))
+
+        self._cells = CardsDict("cel", value)
+
```

Comment:
more than adding a cell one by one, you usually use inp.merge or inp.cells.update(dict). merge should be safe in general, but is update() safe as well?

Reply:
merge was safe indeed as it use the ``__set_item__()`` method at the end
 ```python
cells[key] = value
```
I thought the update method would do the same but it was not apparently. I added a small test and it failed. In the last commit I also re-implemented the update method to force it using the ``__set_item__()`` and it now works as expected